packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -58,6 +59,7 @@ export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
         private readonly trySummarize: (reason: SummarizeReason) => void,
         private readonly minOpsForAttemptOnClose = 50,
     ) {
+        assert(this.configuration.state === "enabled", "Configuration state should be enabled");
```

Comment:
Is this not dangerous? I.e. what happens if we add a new configuration state that is not `enabled` but we would also like to run heuristics? Would it instead be better to check that the state isn't `disabled` or `disableHeuristics`?

Reply:
Yes, I considered that  - if someone tried to add a new condition, we would probably catch really fast. 

But I do not mind adding the or as well.

Reply:
I don't have the full knowledge of what could be expected here. Just wanted to highlight that it could potentially be an issue in the future and might be better to have an opt-in system for these states :)